### PR TITLE
feat: supports new vehicle tables

### PR DESCRIPTION
**Pull request by kyleestes**

### CODE REVIEW

File: DatabaseSchemaReader/CodeGen/ClassWriter.cs [hunk 1]
```diff
@@ -108,52 +108,118 @@ private void WriteWiths()
...
+                    classBuilder.AppendLine($"{propertyName} = {repositoryNameForFfkTable}.{repositoryMethodNameForFfkTable}({repositoryMethodCallParametersForFfkTablePrinted});");
+                    classBuilder.AppendLine("return this;");
                 classBuilder.EndNest();
                 classBuilder.AppendLine("");
             }
+
+
+
+
+            //var fks = CodeWriterUtils.GetWithForeignKeys(table, foreignKeyChild).ToList();
```
Comment (mfalnespn) [inline DatabaseSchemaReader/CodeGen/ClassWriter.cs]: Is there a reason why this code is being saved?  If yes I would add a comment on why.

File: DatabaseSchemaReader/CodeGen/ColumnNameShortnener.cs [hunk 2]
```diff
@@ -0,0 +1,474 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DatabaseSchemaReader.CodeGen
+{
+    public static class ColumnNameShortnener
+    {
+        public static string GetShortName(string columnName)
```
Comment (mfalnespn) [inline DatabaseSchemaReader/CodeGen/ColumnNameShortnener.cs]: Personally I'm not see much value in this method.  I could see stripping off some of the db suffixes like "Indicator" and "Value"

Comment (kyleestes) [reply DatabaseSchemaReader/CodeGen/ColumnNameShortnener.cs]: I wrote this as an experiment, and ended up not using it. I should have deleted it before the PR, sorry about that.

File: DatabaseSchemaReader/CodeGen/ClassWriter.cs (hunk 1 above)
Comment (kyleestes) [reply DatabaseSchemaReader/CodeGen/ClassWriter.cs]: Sorry. This PR was a bit premature -- going to close to address issues.

File: DatabaseSchemaReader/CodeGen/ClassWriter.cs [hunk 3]
```diff
@@ -108,56 +108,55 @@ private void WriteWiths()
 
         public void WriteWith(DatabaseTable foreignKeyChild)
         {
-            foreach (var fk in CodeWriterUtils.GetWithForeignKeys(table, foreignKeyChild))
+            var ffks = CodeWriterUtils.GetWithForeignKeys(table, foreignKeyChild).ToList();
```
Comment (mfalnespn) [inline DatabaseSchemaReader/CodeGen/ClassWriter.cs]: dumb question - what does ffk stand for?  something foreign key but not coming up with the "something"

File: DatabaseSchemaReader/CodeGen/ClassWriter.cs [hunk 4]
```diff
@@ -194,7 +193,13 @@ public void WriteWith(DatabaseConstraint foreignKey)
             }
 
             var s = string.Join(", ", methodCallParameters);
-            classBuilder.AppendLine($"{propertyName} = {CodeWriterUtils.GetRepositoryImplementationName(foreignKey.ReferencedTable(table.DatabaseSchema))}.Get({s});");
+            //classBuilder.AppendLine($"{propertyName} = {CodeWriterUtils.GetRepositoryImplementationName(foreignKey.ReferencedTable(table.DatabaseSchema))}.Get({s});");
```
Comment (mfalnespn) [inline DatabaseSchemaReader/CodeGen/ClassWriter.cs]: Is there a reason for saving this commented out code?

File: DatabaseSchemaReader/CodeGen/CodeWriterUtils.cs [hunk 5]
```diff
@@ -103,17 +103,26 @@ public static IEnumerable<Parameter> AddDbContextParameter(IEnumerable<Parameter
 
         public static string GetGetMethodSignature(DatabaseTable table, CodeWriterSettings codeWriterSettings, IEnumerable<Parameter> methodParameters)
         {
-            return $"{table.NetName} Get({PrintParametersForSignature(methodParameters)})";
+            //return $"{table.NetName} Get({PrintParametersForSignature(methodParameters)})";
```
Comment (mfalnespn) [inline DatabaseSchemaReader/CodeGen/CodeWriterUtils.cs]: Any reason for saving the commented out code?

File: DatabaseSchemaReader/CodeGen/CodeWriterUtils.cs [hunk 6]
```diff
@@ -103,17 +103,26 @@ public static IEnumerable<Parameter> AddDbContextParameter(IEnumerable<Parameter
...
+                return GetMethodParametersForPrimaryKeys(table, codeWriterSettings, byCustomer);
+            }
+
+            return GetMethodParametersForUniqueConstraint(table, codeWriterSettings, byCustomer);
         }
 
         public static string GetGetListMethodSignature(DatabaseTable table, CodeWriterSettings codeWriterSettings, IEnumerable<Parameter> methodParameters)
         {
-            return $"IEnumerable<{table.NetName}> GetList({PrintParametersForSignature(methodParameters)})";
+            //return $"IEnumerable<{table.NetName}> GetList({PrintParametersForSignature(methodParameters)})";
```
Comment (mfalnespn) [inline DatabaseSchemaReader/CodeGen/CodeWriterUtils.cs]: same question

**Comment by kyleestes** [comment]
Not a dumb question! Ffk stands for foreign foreign key. Maybe I need to
make that variable name just say that

**Pull request merged**